service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsHttpEndpoint.java

```diff
@@ -41,6 +44,8 @@
     // path parameters for capturing parts of the URI path
     private static final String PARAM_TYPE = "type";
     private static final String PARAM_AUTH_ID = "auth_id";
+    private static final String DEFAULT_HASH_FUNCTION = System.getProperty("pwd-hash-function",
```

Comment:
we usually do not read environment variables directly but instead have them mapped to config object properties

Reply:
That would require to define a new ConfigProperties class, right? It feels a little bit like an overkill. 
An alternative idea would be to let the client request a hash function (and salt for shaX), which would be even closer to the current syntax. Example:

```
$ curl -i -X POST -H 'Content-Type: application/json' --data-binary '{
    "device-id": "4720",
    "type": "hashed-password",
    "auth-id": "sensor20",
    "secrets": [{
        "hash-function" : "bcrypt",
        "pwd-plain": "mylittlesecret"
    }]
  }' http://localhost:28080/credentials/DEFAULT_TENANT```

WDYT?

Reply:
letting the user select a hash function sounds good, but we shouldn't allow the client to specify the salt. That should be handled on the server side where we can guarantee the use of a secure random number generator.